core/domain/suggestion_jobs_one_off.py

```diff
@@ -0,0 +1,108 @@
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS-IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""One-off jobs for suggestions."""
+
+from __future__ import absolute_import  # pylint: disable=import-only-modules
+from __future__ import unicode_literals  # pylint: disable=import-only-modules
+
+import ast
+import logging
+
+from core import jobs
+from core.domain import html_validation_service
+from core.domain import suggestion_registry
+from core.domain import suggestion_services
+from core.platform import models
+
+(suggestion_models,) = models.Registry.import_models([models.NAMES.suggestion])
+
+
+class SuggestionMathRteAuditOneOffJob(jobs.BaseMapReduceOneOffJobManager):
+    """Job that checks for existence of math components in the suggestions."""
+
+    @classmethod
+    def entity_classes_to_map_over(cls):
+        return [suggestion_models.GeneralSuggestionModel]
+
+    @staticmethod
+    def map(item):
+        html_string_list = []
+        suggestion = suggestion_services.get_suggestion_from_model(item)
+        html_string_list = (
+            html_string_list + suggestion.get_all_html_content_strings())
+        html_string = ''.join(html_string_list)
+        if html_validation_service.check_for_math_component_in_html(
+                html_string):
+            yield ('Suggestion with Math', item.id)
+
+    @staticmethod
+    def reduce(key, values):
+        yield (
+            '%d suggestions have Math components in them with IDs.' %
+            len(values), values)
+
+
+class SuggestionMathMigrationOneOffJob(jobs.BaseMapReduceOneOffJobManager):
+    """A one-time job that can be used to migrate the Math components in the
+    suggestions to the new Math Schema.
+    """
+    _ERROR_KEY = 'validation_error'
+
+    @classmethod
+    def entity_classes_to_map_over(cls):
+        return [suggestion_models.GeneralSuggestionModel]
+
+    @staticmethod
+    def map(item):
+        suggestion = suggestion_services.get_suggestion_by_id(item.id)
+        try:
+            suggestion.validate()
+        except Exception as e:
+            logging.error(
+                'Suggestion %s failed validation: %s' % (item.id, e))
+            yield (
+                SuggestionMathMigrationOneOffJob._ERROR_KEY,
+                'Suggestion %s failed validation: %s' % (item.id, e))
+            return
+        suggestion_domain_class = (
+            suggestion_registry.SUGGESTION_TYPES_TO_DOMAIN_CLASSES[
+                suggestion.suggestion_type])
+        suggestion.change = (
+            suggestion_domain_class.convert_html_in_suggestion(
+                suggestion.change.to_dict(),
```

Comment:
Changed the docstring and function name

Reply:
Can't you make this a function on the object itself? You are taking a part of the object and setting a part of the object, so I think it would be more sensible to make it a function on the object instead of a classmethod.

Reply:
@nithusha21, we had discussed this point before pls refer [here](https://github.com/oppia/oppia/pull/9390/#discussion_r432852227).
Also, this function doesn't modify the object, it just takes a dict, converts a dict and returns a change object.
It doesn't modify the object.